src/OpenTelemetry/Internal/EnvironmentVariables/EnvironmentVariablesConfigurationProvider.cs

```diff
@@ -0,0 +1,122 @@
+// <auto-generated /> (Turns off StyleCop analysis in this file.)
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+#nullable enable
+
+using System;
+using System.Collections;
+using System.Collections.Generic;
+
+namespace Microsoft.Extensions.Configuration.EnvironmentVariables
+{
+    /// <summary>
+    /// An environment variable based <see cref="ConfigurationProvider"/>.
+    /// </summary>
+    internal sealed class EnvironmentVariablesConfigurationProvider : ConfigurationProvider
+    {
+        private const string MySqlServerPrefix = "MYSQLCONNSTR_";
+        private const string SqlAzureServerPrefix = "SQLAZURECONNSTR_";
+        private const string SqlServerPrefix = "SQLCONNSTR_";
+        private const string CustomConnectionStringPrefix = "CUSTOMCONNSTR_";
```

Comment:
@alanwest @cijothomas I was just chatting with @reyang about this. So the new feature we are supporting in 1.4.0 is we will look for environment variables using IConfiguration. This code is only ever used for Sdk.Create* style to build IConfiguration from environment variables as backwards compatibility for users without hosting. So strictly speaking, we do not need this connection string logic. We don't support it in 1.3.0 and we don't need to in 1.4.0. Thinking we should remove this and slim it down as a follow-up.

Reply:
https://github.com/open-telemetry/opentelemetry-dotnet/pull/4106